### DISCUSSION TITLE
Does JSONSchema support assigning a function to a property?

### DISCUSSION THREAD

**Discussion by reselbob** [json-schema-org/community#65] [Q&A] [answered]
I want to be able to define functions with parameters and return type for my JavaScript object specification. I am hoping to be able to do this using JSONSchema. But, I can't seem to figure out how to do it.

I am trying to do something like the following:

```
{
  "$schema": "https://json-schema.org/draft/2020-12/schema",
  "$id": "https://example.com/product.schema.json",
  "title": "Product",
  "description": "A JavaScript object that gets a product from Acme's catalog",
  "type": "object",
  "properties": {
    "getProduct": {
      "description": "returns a product object when a product id is provided as a parameter",
      "type": "function",
      "parameters": ["id": uuid],
      "returnType": object
    }
  },
}
```
Is it possible to do something liek this? Triple thanks in advance.

**Comment by SquareByte** [comment, marked as answer]
As (JavaScript) functions are not part of the JSON specification, it would be out of scope for JSONSchema.

You can of course define your own schema or your own vocabulary along the lines of what you example code shows.

**Comment by gregsdennis** [comment]
Question was already answered on [StackOverflow](https://stackoverflow.com/questions/69476002/how-to-a-specify-a-property-that-is-a-function-in-json-schema).

**Comment by reselbob** [comment]
Thanks for the update. I am accustomed to using OpenAPI to specify a REST API, so "functions" are sorta built in to the specification format. I was hoping to take a similar route under JSON Schema. Of course TypeScript supports interfaces, which is an implicit specification for an object. We're trying to implement a single source of truth system in which both developers and documentation creators work from the same source in order to create code and also to create documentation.

**Comment by karenetheridge** [comment]
https://json-schema.org/implementations.html#code-generation

**Reply by karenetheridge** [reply]
also see https://github.com/json-schema-org/vocab-idl which is attempting to put this into a specification.
